cluster/security_config.go

```diff
@@ -1,12 +1,8 @@
 package cluster
 
 type SecurityConfig struct {
-	Username string
-	Password string
-}
-
-func NewSecurityConfig() SecurityConfig {
-	return SecurityConfig{}
+	Username string `json:",omitempty"`
```

Comment:
How will this `SecurityConfig` look like when we add more security options in the future? (like `KerberosIdentityConfig` from Java)

Reply:
I didn't think thoroughly about that but we would probably put it under `SecurityConfig`. I guess putting `Username` and `Password` in something like `SecurityConfig.CredentialsConfig` would make more sense in that case 🤔 🤔 🤔

Reply:
I've moved it to `config.Cluster.Security.Credentials`: https://github.com/hazelcast/hazelcast-go-client/pull/585/commits/f3ad108cdecae243f1381d19af33c0723e4d4a96